core-backend/src/memory.rs

```diff
@@ -16,36 +16,121 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! Work with WASM program memory in backends.
+//! sp-sandbox extensions for memory.
 
 use crate::{
-    runtime::RunFallibleError, BackendExternalities, BackendSyscallError, TrapExplanation,
+    state::HostState, BackendExternalities, BackendSyscallError, RunFallibleError, TrapExplanation,
     UndefinedTerminationReason, UnrecoverableMemoryError,
 };
 use alloc::vec::Vec;
-use core::{
-    fmt::Debug,
-    marker::PhantomData,
-    mem,
-    mem::{size_of, MaybeUninit},
-    result::Result,
-    slice,
-};
+use codec::{Decode, DecodeAll, MaxEncodedLen};
+use core::{marker::PhantomData, mem, mem::MaybeUninit, slice};
 use gear_core::{
     buffer::{RuntimeBuffer, RuntimeBufferSizeError},
-    memory::{Memory, MemoryError, MemoryInterval},
+    env::Externalities,
+    memory::{HostPointer, Memory, MemoryError, MemoryInterval},
+    pages::{PageNumber, PageU32Size, WasmPage},
 };
 use gear_core_errors::MemoryError as FallibleMemoryError;
-use num_enum::{IntoPrimitive, TryFromPrimitive};
-use scale_info::scale::{Decode, DecodeAll, MaxEncodedLen};
-
-/// Memory access error during sys-call that lazy-pages have caught.
-/// 0 index is reserved for an ok result.
-#[derive(Debug, Clone, IntoPrimitive, TryFromPrimitive)]
-#[repr(u8)]
-pub enum ProcessAccessError {
-    OutOfBounds = 1,
-    GasLimitExceeded = 2,
+use gear_lazy_pages_common::ProcessAccessError;
+use gear_sandbox::{
+    default_executor::{Caller, Store},
+    SandboxMemory,
+};
+
+pub type DefaultExecutorMemory = gear_sandbox::default_executor::Memory;
```

Comment:
Since we have no solid choose of it, we can just reimport Memory itself?

Reply:
We can't since we also have `Memory` trait in core